test/Microsoft.Windows.CsWin32.Tests/GeneratorTests.cs

```diff
@@ -602,6 +602,40 @@ public void HandleTypeDefsUseIntPtrAsFieldType(string handleType)
         Assert.Equal(nameof(IntPtr), Assert.IsType<IdentifierNameSyntax>(field.Declaration.Type).Identifier.ValueText);
     }
 
+    [Fact]
+    public void HasGeneratedCodeAttribute()
+    {
+        this.generator = this.CreateGenerator();
+        Assert.True(this.generator.TryGenerate("IDebugDocumentInfo", CancellationToken.None));
+        Assert.True(this.generator.TryGenerate("HANDLE", CancellationToken.None));
+        Assert.True(this.generator.TryGenerate("INPUT_RECORD", CancellationToken.None));
+        Assert.True(this.generator.TryGenerate("GetTickCount", CancellationToken.None));
+        Assert.True(this.generator.TryGenerate("ACTIVATE_KEYBOARD_LAYOUT_FLAGS", CancellationToken.None));
+        this.CollectGeneratedCode(this.generator);
+        this.AssertNoDiagnostics();
+
+        InterfaceDeclarationSyntax interfaceDecl = Assert.IsType<InterfaceDeclarationSyntax>(this.FindGeneratedType("IDebugDocumentInfo").Single());
+        Assert.Contains(interfaceDecl.AttributeLists, al => al.Attributes.Any(a => a.Name.ToString().Contains("GeneratedCode")));
+
+        StructDeclarationSyntax handle = Assert.IsType<StructDeclarationSyntax>(this.FindGeneratedType("HANDLE").Single());
+        Assert.Contains(handle.AttributeLists, al => al.Attributes.Any(a => a.Name.ToString().Contains("GeneratedCode")));
+
+        StructDeclarationSyntax structDecl = Assert.IsType<StructDeclarationSyntax>(this.FindGeneratedType("INPUT_RECORD").Single());
+        Assert.Contains(structDecl.AttributeLists, al => al.Attributes.Any(a => a.Name.ToString().Contains("GeneratedCode")));
+
+        IEnumerable<ClassDeclarationSyntax> pInvokeClass = this.FindGeneratedType("PInvoke").OfType<ClassDeclarationSyntax>();
+        Assert.Contains(pInvokeClass, c => c.AttributeLists.Any(al => al.Attributes.Any(a => a.Name.ToString().Contains("GeneratedCode"))));
+
+        EnumDeclarationSyntax enumDecl = Assert.IsType<EnumDeclarationSyntax>(this.FindGeneratedType("ACTIVATE_KEYBOARD_LAYOUT_FLAGS").Single());
+        Assert.Contains(enumDecl.AttributeLists, al => al.Attributes.Any(a => a.Name.ToString().Contains("GeneratedCode")));
+
+        ClassDeclarationSyntax arrayExtensions = Assert.IsType<ClassDeclarationSyntax>(this.FindGeneratedType("InlineArrayIndexerExtensions").Single());
```

Comment:
InlineArrayIndexerExtensions and FriendlyOverloadExtensions aren't generated, I need to find an associated token which will generate them.

Reply:
@AArnott this is the only outstanding problem with this issue AFAIK. Could you please give me some guidance on how I can test for this?

Reply:
I found the token to add. But now the test fails because the attribute is added above the xml doc comment for templated structs. I'm taking a look to see if I can fix it now.

Reply:
I had the same issue elsewhere and had to make sure the order of function calls was correct.